src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -257,13 +292,111 @@ class CassandraJournal(cfg: Config) extends AsyncWriteJournal with CassandraReco
   private def persistentToByteBuffer(p: PersistentRepr): ByteBuffer =
     ByteBuffer.wrap(serialization.serialize(p).get)
 
-  def persistentFromByteBuffer(b: ByteBuffer): PersistentRepr = {
+  private def persistentFromByteBuffer(b: ByteBuffer): PersistentRepr = {
     serialization.deserialize(Bytes.getArray(b), classOf[PersistentRepr]).get
   }
 
   private case class SerializedAtomicWrite(persistenceId: String, payload: Seq[Serialized])
   private case class Serialized(sequenceNr: Long, serialized: ByteBuffer, tags: Set[String])
   private case class PartitionInfo(partitionNr: Long, minSequenceNr: Long, maxSequenceNr: Long)
+
+  /**
+   * Iterator over messages, crossing partition boundaries.
+   */
+  private class MessageIterator(persistenceId: String, fromSequenceNr: Long, toSequenceNr: Long, max: Long) extends Iterator[PersistentRepr] {
```

Comment:
Is this only used for `readLowestSequenceNr` now? then this should be optimized for sequenceNr (eg. not deserialize the message). If that is not part of this PR, please add a FIXME.